arroyo-sql/src/pipeline.rs

```diff
@@ -473,13 +488,50 @@ impl<'a> SqlPipelineBuilder<'a> {
         let input = self.insert_sql_plan(&filter.input)?;
         let struct_def = input.return_type();
         let ctx = self.ctx(&struct_def);
-        let predicate = ctx.compile_expr(&filter.predicate)?;
+        let mut predicate = ctx.compile_expr(&filter.predicate)?;
+
+        Self::assert_no_unnest("where", &mut predicate)?;
```

Comment:
Would this be clearer with something like
```
if Self::has_unnest(&mut predicate) {
    bail!("can't have unnest in WHERE clause");
}

Reply:
given this appears in a bunch of places, I preferred to move the bail call into the function so that I wasn't repeating it over and over again. To me at least, having `assert` and the `?` makes it clear that this can early-exit.